src/Domain/Structure/Neighbors.cpp

```diff
@@ -12,19 +13,56 @@
 #include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/StdHelpers.hpp"
 
+template <size_t VolumeDim, typename IdType>
+Neighbors<VolumeDim, IdType>::Neighbors(
+    std::unordered_set<IdType> ids,
+    std::unordered_map<size_t, OrientationMap<VolumeDim>> orientations,
+    const bool are_conforming)
+    : ids_(std::move(ids)),
+      orientations_(std::move(orientations)),
+      are_conforming_(are_conforming) {
+  ASSERT(std::ranges::none_of(orientations_,
+                              [](const auto& orientation) {
+                                return orientation.second ==
+                                       OrientationMap<VolumeDim>{};
+                              }),
+         "Cannot use a default-constructed OrientationMap in Neighbors.");
+}
+
 template <size_t VolumeDim, typename IdType>
 Neighbors<VolumeDim, IdType>::Neighbors(std::unordered_set<IdType> ids,
                                         OrientationMap<VolumeDim> orientation)
-    : ids_(std::move(ids)), orientation_(std::move(orientation)) {
-  ASSERT(orientation_ != OrientationMap<VolumeDim>{},
+    : ids_(std::move(ids)) {
+  ASSERT(orientation != OrientationMap<VolumeDim>{},
          "Cannot use a default-constructed OrientationMap in Neighbors.");
+  if constexpr (std::is_same_v<size_t, IdType>) {
+    orientations_.emplace(*ids_.begin(), std::move(orientation));
+  } else {
+    orientations_.emplace(ids_.begin()->block_id(), std::move(orientation));
+  }
 }
 
 template <size_t VolumeDim, typename IdType>
 Neighbors<VolumeDim, IdType>::Neighbors(const IdType id,
                                         OrientationMap<VolumeDim> orientation)
     : Neighbors(std::unordered_set{std::move(id)}, std::move(orientation)) {}
 
+template <size_t VolumeDim, typename IdType>
+const OrientationMap<VolumeDim>& Neighbors<VolumeDim, IdType>::orientation(
+    const IdType& id) const {
+  if constexpr (std::is_same_v<size_t, IdType>) {
+    return orientations_.at(id);
+  } else {
+    return orientations_.at(id.block_id());
+  }
+}
+
+template <size_t VolumeDim, typename IdType>
+void Neighbors<VolumeDim, IdType>::set_ids_to(
+    const std::unordered_set<IdType> new_ids) {
+  ids_ = std::move(new_ids);
```

Comment:
add `ASSERT` that the blocks being added have orientation maps stored in the class like the dox says is required

Reply:
was done in the next commit